### PR TITLE
[ACS-3742] Layout changes for workspace

### PR DESCRIPTION
**Pull request by Yasa-Nataliya**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [x] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

image

Layout changes for ACA according to the ACS-3742 jira ticket.
Depends on: https://github.com/Alfresco/alfresco-ng2-components/pull/8256

- improved evaluators for `Create` and `Upload` (https://alfresco.atlassian.net/browse/ACS-4712)

**Does this PR introduce a breaking change?** (check one with "x")

> - [x] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

- custom header image path is deprecated
- custom header text color is deprecated 
- custom header color is deprecated

**Other information**:

https://alfresco.atlassian.net/browse/ACS-3742
https://alfresco.atlassian.net/browse/ACS-4708
https://alfresco.atlassian.net/browse/ACS-4712

### CODE REVIEW

File: projects/aca-about/src/lib/about.component.scss [hunk 1]
```diff
@@ -2,3 +2,13 @@ adf-about {
   padding: 10px;
   width: 100%;
 }
+
+.app-about-container {
+  position: fixed;
```
Comment (DenysVuika) [inline projects/aca-about/src/lib/about.component.scss:7]: this is incorrect, we should not require fixed positioning for proper layouting; width and height to 100% should do the trick

File: projects/aca-shared/src/lib/services/app.service.ts [hunk 2]
```diff
@@ -65,6 +65,9 @@ export class AppService implements OnDestroy {
   private ready: BehaviorSubject<boolean>;
   ready$: Observable<boolean>;
 
+  public hideSidenav = new BehaviorSubject<boolean>(false);
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/services/app.service.ts:68]: should be private ? or why you have secondary public below?

File: projects/aca-shared/src/lib/services/app.service.ts [hunk 3]
```diff
@@ -65,6 +65,9 @@ export class AppService implements OnDestroy {
...
+  cast = this.hideSidenav.asObservable();
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/services/app.service.ts:69]: why it is called "cast" and why "cast" is responsible for the hiding the side nav? you probably need making "hideSidenav" private and making this one public and called 'hideSidenav$'

File: projects/aca-shared/src/lib/services/app.service.ts [hunk 4]
```diff
@@ -116,6 +119,10 @@ export class AppService implements OnDestroy {
     );
   }
 
+  getSidenavValue(str: boolean) {
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/services/app.service.ts:122]: this is called "get value" but it is not getting anything, just emitting "next"... why parameter is called "str" and and has Boolean type? can you explain the meaning of all this here, thanks

File: projects/aca-shared/src/lib/components/page-layout/page-layout.module.ts [hunk 5]
```diff
@@ -29,9 +29,10 @@ import { PageLayoutErrorComponent } from './page-layout-error.component';
 import { PageLayoutHeaderComponent } from './page-layout-header.component';
 import { PageLayoutComponent } from './page-layout.component';
 import { CommonModule } from '@angular/common';
+import { CoreModule } from '@alfresco/adf-core';
 
 @NgModule({
-  imports: [CommonModule],
+  imports: [CommonModule, CoreModule.forChild()],
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/components/page-layout/page-layout.module.ts:35]: do you need the whole CoreModule here or maybe some particular modules can work?

File: projects/aca-shared/src/lib/components/page-layout/page-layout.component.ts [hunk 6]
```diff
@@ -36,4 +37,20 @@ import { Component, ViewEncapsulation, ChangeDetectionStrategy, Input } from '@a
 export class PageLayoutComponent {
   @Input()
   hasError = false;
+  hideSidenav: boolean;
+
+  constructor(private appService: AppService, private ref: ChangeDetectorRef) {
+    setInterval(() => {
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/components/page-layout/page-layout.component.ts:43]: this is absolutely wrong!!!

File: projects/aca-shared/src/lib/components/page-layout/page-layout.component.ts [hunk 7]
```diff
@@ -36,4 +37,20 @@ import { Component, ViewEncapsulation, ChangeDetectionStrategy, Input } from '@a
 export class PageLayoutComponent {
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/components/page-layout/page-layout.component.ts:37]: missing OnInit interface

File: projects/aca-shared/src/lib/components/page-layout/page-layout.component.ts [hunk 8]
```diff
@@ -36,4 +37,20 @@ import { Component, ViewEncapsulation, ChangeDetectionStrategy, Input } from '@a
...
+    });
+  }
+
+  ngOnInit() {
+    this.appService.cast.subscribe((data) => (this.hideSidenav = data));
+  }
+
+  toggleClick() {
+    this.hideSidenav = !this.hideSidenav;
+    this.appService.getSidenavValue(this.hideSidenav);
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/components/page-layout/page-layout.component.ts:54]: this is absolutely confusing, get value and do what with that value?

File: projects/aca-shared/src/lib/components/page-layout/page-layout.component.html [hunk 9]
```diff
@@ -1,3 +1,11 @@
-<ng-content select="aca-page-layout-header"></ng-content>
-<ng-content select="aca-page-layout-error" *ngIf="hasError"></ng-content>
-<ng-content select="aca-page-layout-content" *ngIf="!hasError"></ng-content>
+<div class="aca-content-container" [ngClass]="{'aca-page-layout-container' : hideSidenav}">
+    <div class="aca-content-header">
+        <div class="aca-menu-icon" [ngClass]="{'aca-display-menu' : !hideSidenav}" (click)="toggleClick()">
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/components/page-layout/page-layout.component.html:3]: even github linter suggests that "click must be accompanied by either keyup, keydown or keypress event for accessibility" please run linting of your code before creating a PR

**Review by DenysVuika — CHANGES_REQUESTED** [review]

**Comment by DenysVuika** [comment]
Material Icons already have "cog" icon, we don't need another copy (`projects/aca-content/assets/svg/icons/settings.svg`)

**Comment by DenysVuika** [comment]
You have strange UI in the about section:
image

**Comment by DenysVuika** [comment]
Username alignment still not fixed
image

**Comment by DenysVuika** [comment]
User profile "Edit" buttons look ugly and text looks not aligned properly 
image

**Comment by DenysVuika** [comment]
Please run "ng lint" and fix the issues... it won't allow you to run the rest of the tests before you do that

File: projects/aca-content/src/lib/components/sidenav/sidenav.component.html [hunk 10]
```diff
@@ -1,32 +1,44 @@
-<div class="sidenav">
+<div class="sidenav" *ngIf="!hideSidenav">
   <ng-container [ngSwitch]="mode">
-    <div class="section action-menu" [ngClass]="'section--' + mode">
-      <app-main-action [expanded]="mode === 'expanded'"></app-main-action>
-      <app-create-menu [expanded]="mode === 'expanded'"></app-create-menu>
+    <div class="sidenav-header">
+      <div class="sidenav-header-title">
+        <div class="sidenav-header-title-logo" (click)="toggleClick()">
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/sidenav/sidenav.component.html:5]: this shoube be mat-button-icon, not the clickable div, otherwise you are going to get a11y issues

**Comment by DenysVuika** [comment]
Sidebar has issue with long text and localisation, the long text is going out of the view 
image

File: projects/aca-about/src/lib/about.component.ts [hunk 11]
```diff
@@ -58,6 +61,7 @@ export class AboutComponent implements OnInit {
     if (this.authService.isEcmLoggedIn()) {
       this.setECMInfo();
     }
+    this.appServices.cast.subscribe((data) => (this.hideSidenav = data));
```
Comment (DenysVuika) [inline projects/aca-about/src/lib/about.component.ts:64]: this can be done by observable and async pipe, similar to extensions.... please check other files or read some docs on the async pipes.... You should also have "onDestroy" to properly unsubscribe... you can find plenty of examples in the existing code

File: projects/aca-content/assets/i18n/en.json [hunk 12]
```diff
@@ -102,7 +103,20 @@
...
         },
+        "HEADER": {
+            "BUTTONS": {
+                "CREATE": "Create",
+                "UPLOAD": "Upload"
+            }
+        },
         "BROWSE": {
+            "FILE": {
+                "TITLE": "Files",
```
Comment (DenysVuika) [inline projects/aca-content/assets/i18n/en.json:114]: I don't see any code using this i18n, where should it go?

File: projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.html [hunk 13]
```diff
@@ -0,0 +1,35 @@
+<button mat-icon-button [matMenuTriggerFor]="menu" aria-label="matMenu" class="app-menu-button">
+    <mat-icon svgIcon="menu" title="{{'APP.TOOLTIPS.OPTIONS_SETTINGS' | translate}}"></mat-icon>
+</button>
+<mat-menu #menu="matMenu" xPosition="before">
+    <div mat-menu-item class="app-menu-user-details" [routerLink]="['/profile']" title="{{'APP.TOOLTIPS.MY_PROFILE' | translate}}">
+        <button class="app-menu-user-details-button" aria-label="matMenu">
+            <mat-icon svgIcon="avatar"></mat-icon>
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.html:7]: this is incorrect,  according to existing ADF implementation this should be initials of the user, for example 'SA' for Super Admin, 'A' for admin, and so on... it's not the 'A'  icon for every user

File: projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.html [hunk 14]
```diff
@@ -0,0 +1,35 @@
+<button mat-icon-button [matMenuTriggerFor]="menu" aria-label="matMenu" class="app-menu-button">
+    <mat-icon svgIcon="menu" title="{{'APP.TOOLTIPS.OPTIONS_SETTINGS' | translate}}"></mat-icon>
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.html:2]: this makes zero sense.... the menu says "A" letter, what does that mean? the initials of the Admin? why every user should get static "A" image?

**Comment by DenysVuika** [comment]
You have icons committed in 2 places: app assets and library assets, that is not correct

**Comment by DenysVuika** [comment]
About screen, once you collapse the sidebar, it's not visible anymore.... insane
image

File: projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.html [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+    </div>
+    <mat-divider></mat-divider>
+    <button mat-menu-item data-automation-id="language-menu-button" [matMenuTriggerFor]="langMenu">
+        <mat-icon>language</mat-icon>
+        <span>{{ 'APP.LANGUAGE' | translate }}</span>
+    </button>
+    <mat-menu #langMenu="matMenu" class="adf-menu">
+        <adf-language-menu></adf-language-menu>
+    </mat-menu>
+    <button mat-menu-item data-automation-id="about" [routerLink]="['/about']">
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.html:17]: this is absolutely incorrect, if you check existing code, About and Settings must be coming from the plugins dynamically, not hardcoded like you do

File: projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.html [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+        <mat-icon>info</mat-icon>
+        {{ 'APP.BROWSE.ABOUT.TITLE' | translate }}
+    </button>
+    <button mat-menu-item data-automation-id="user-language-menu" [routerLink]="['/settings']">
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.html:21]: copy-pasted code, wrong automation id `data-automation-id="user-language-menu"` for the `settings`

**Comment by DenysVuika** [comment]
Bug: About screen, clicking Files expand/collapse redirects to personal files

File: projects/aca-content/src/lib/components/header/header.component.html [hunk 17]
```diff
@@ -1,19 +0,0 @@
-<adf-layout-header
-  [logo]="logo$ | async"
-  [redirectUrl]="landingPage"
-  [tooltip]="appName$ | async"
-  [color]="headerColor$ | async"
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/header/header.component.html:5]: this requires cleaning all the documentation related to these properties you remove

File: projects/aca-shared/src/lib/components/page-layout/page-layout.component.html [hunk 18]
```diff
@@ -1,3 +1,11 @@
-<ng-content select="aca-page-layout-header"></ng-content>
-<ng-content select="aca-page-layout-error" *ngIf="hasError"></ng-content>
-<ng-content select="aca-page-layout-content" *ngIf="!hasError"></ng-content>
+<div class="aca-content-container" [ngClass]="{'aca-page-layout-container' : hideSidenav}">
+    <div class="aca-content-header">
+        <div class="aca-menu-icon" [ngClass]="{'aca-display-menu' : !hideSidenav}" (click)="toggleClick()" (keypress)="toggleClick()">
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/components/page-layout/page-layout.component.html:3]: your class "aca-display-menu" actually hides content, so you say hide content when hideSidenav = false ? this is just stupid

File: projects/aca-content/assets/i18n/en.json [hunk 19]
```diff
@@ -1,6 +1,7 @@
 {
     "APP": {
       "COPYRIGHT": "© 2017 - 2020 Alfresco Software, Inc. All rights reserved.",
+      "TITLE": "Workspace",
```
Comment (DenysVuika) [inline projects/aca-content/assets/i18n/en.json:4]: we have https://github.com/Alfresco/alfresco-content-app/blob/develop/app/src/app.config.json.tpl#L28 for that

File: projects/aca-content/src/lib/components/sidenav/sidenav.component.html [hunk 20]
```diff
@@ -1,32 +1,44 @@
-<div class="sidenav">
+<div class="sidenav" *ngIf="!hideSidenav">
   <ng-container [ngSwitch]="mode">
-    <div class="section action-menu" [ngClass]="'section--' + mode">
-      <app-main-action [expanded]="mode === 'expanded'"></app-main-action>
-      <app-create-menu [expanded]="mode === 'expanded'"></app-create-menu>
+    <div class="sidenav-header">
+      <div class="sidenav-header-title">
+        <div class="sidenav-header-title-logo" (click)="toggleClick()" (keypress)="toggleClick()">
+          <mat-icon svgIcon="workspace"
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/sidenav/sidenav.component.html:6]: the icon, logo and navigation paths are incorrectly set... if you check the original component that you rework, it has configuration support:
```
this.appName$ = store.select(getAppName);
    this.logo$ = store.select(getLogoPath);
    this.landingPage = this.appConfigService.get('landingPage', '/personal-files');
```

File: projects/aca-content/src/lib/components/header-actions/header-actions.component.scss [hunk 21]
```diff
@@ -0,0 +1,55 @@
...
+    letter-spacing: 0.25px;
+    font-weight: 500;
+    font-size: 14px;
+    font-style: normal;
+    border-radius: 6px;
+    border: none;
+  }
+
+  .aca-create-button {
+    width: 71px !important;
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/header-actions/header-actions.component.scss:34]: why this is 71px and important? what if it's another language and it needs more space?

File: projects/aca-content/src/lib/components/header-actions/header-actions.component.scss [hunk 22]
```diff
@@ -0,0 +1,55 @@
...
+    height: 32px;
+    background-color: var(--theme-create-button-background-color);
+    color: var(--theme-create-button-text-color);
+    text-overflow: ellipsis;
+    display: flex;
+    padding: 0;
+  }
+
+  .aca-upload-button {
+    width: 73px !important;
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/header-actions/header-actions.component.scss:45]: why this is 73px and important? what if it's another language and it needs more space?

File: projects/aca-content/src/lib/components/search/search-input/search-input.component.scss [hunk 23]
```diff
@@ -54,6 +74,8 @@ mat-checkbox {
   background-color: $search-background;
   border-radius: $search-border-radius;
   height: $search-height;
+  margin-bottom: 0 !important;
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/search/search-input/search-input.component.scss:77]: not sure why this is needed

File: projects/aca-content/src/lib/components/view-profile/view-profile.component.html [hunk 24]
```diff
@@ -1,10 +1,9 @@
-<div class="app-profile-container">
+<div class="app-profile-container" [ngClass]="{'app-view-profile-container' : hideSidenav === true}">
   <div class="app-profile-row">
     <div class="app-profile-title">
       <mat-icon class="app-profile-icon" (click)="navigateToPersonalFiles()" id="backButton">arrow_back</mat-icon>
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/view-profile/view-profile.component.html:4]: should be navigating to the default path driven by the configuration, we already had this support before

File: projects/aca-content/src/lib/components/header-actions/header-actions.component.ts [hunk 25]
```diff
@@ -0,0 +1,88 @@
...
+  canShowUploadButton(): boolean {
+    if (this.isPersonalFilesRoute()) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  canShowSearchSeparator(): boolean {
+    if (this.isPersonalFilesRoute() || this.isFavoriteLibrariesRoute() || this.isLibrariesRoute()) {
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/header-actions/header-actions.component.ts:82]: you need to show separator if there's any other buttons there, not because it's personal files or favorites

File: projects/aca-content/src/lib/components/header-actions/header-actions.component.ts [hunk 26]
```diff
@@ -0,0 +1,88 @@
...
+    super.ngOnInit();
+  }
+
+  ngOnDestroy() {
+    this.store.dispatch(new SetCurrentFolderAction(null));
+    super.ngOnDestroy();
+  }
+
+  isPersonalFilesRoute(): boolean {
+    return this.router.url.includes('/personal-files');
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/header-actions/header-actions.component.ts:54]: even with this childish solution to have multiple methods to check the router url... you could have an array instead and loop through it...

File: projects/aca-content/src/lib/components/search/search-input/search-input.component.html [hunk 27]
```diff
@@ -20,7 +21,18 @@
     <div matSuffix class="app-suffix-search-icon-wrapper">
       <mat-icon>arrow_drop_down</mat-icon>
     </div>
-  </mat-form-field>
+
+    <button mat-icon-button matSuffix class="app-suffix-search-icon-wrapper app-close-icon" (click)="exitSearch()" (keypressk)="exitSearch()">
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/search/search-input/search-input.component.html:25]: typo in the `keypressk`

File: projects/aca-content/src/lib/components/header-actions/header-actions.component.ts [hunk 28]
```diff
@@ -0,0 +1,88 @@
...
+  }
+
+  isFavoriteLibrariesRoute(): boolean {
+    return this.router.url.includes('/favorite/libraries');
+  }
+
+  isLibrariesRoute(): boolean {
+    return this.router.url.includes('/libraries');
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/header-actions/header-actions.component.ts:62]: this will return true for `/favorite/libraries` too, as you are using "includes"

File: projects/aca-content/src/lib/components/header/header.component.spec.ts [hunk 29]
```diff
@@ -85,38 +82,11 @@ describe('AppHeaderComponent', () => {
...
-  describe('Search input', () => {
-    beforeEach(() => {
-      localStorage.clear();
-    });
-
-    afterEach(() => {
-      localStorage.clear();
-    });
-
-    it('should search be present when contentService is enabled', () => {
```
Comment (MichalKinas) [inline projects/aca-content/src/lib/components/header/header.component.spec.ts:107]: Why did you removed this unit tests? Nothing changed with aca search within this component as far as I can see

File: projects/aca-content/src/lib/components/search/search-input-control/search-input-control.component.scss [hunk 30]
```diff
@@ -6,6 +6,7 @@ $top-margin: 12px;
   font-size: 16px;
   padding-left: 15px;
   box-sizing: border-box;
+  margin-bottom: 12px !important;
```
Comment (MichalKinas) [inline projects/aca-content/src/lib/components/search/search-input-control/search-input-control.component.scss:9]: Is the `important` required?

File: projects/aca-content/src/lib/components/search/search-input.service.ts [hunk 31]
```diff
@@ -23,17 +23,33 @@
...
+  isSearchRoute(): boolean {
+    return this.router.url.includes('/search');
+  }
+
+  saveRoute(route: string) {
+    this.savedRoute = route;
+  }
+
+  exitSearch() {
+    if (this.savedRoute.length > 0) {
```
Comment (MichalKinas) [inline projects/aca-content/src/lib/components/search/search-input.service.ts:46]: What if someone will call `saveRoute` with `null`? I think you should add null check for `savedRoute` here

File: projects/aca-content/src/lib/components/search/search-input.service.ts [hunk 32]
```diff
@@ -23,17 +23,33 @@
...
-@Component({
-  selector: 'aca-sidenav-wrapper',
-  templateUrl: './sidenav-wrapper.component.html'
+@Injectable({
+  providedIn: 'root'
 })
-export class SidenavWrapperComponent {
-  @Input()
-  data: { mode?: 'collapsed' | 'expanded' } = {};
+export class SearchInputService {
```
Comment (MichalKinas) [inline projects/aca-content/src/lib/components/search/search-input.service.ts:32]: Also name of this service is a bit confusing since it has nothing to do with search input, it just manages the routes and navigates to search

File: projects/aca-content/src/lib/components/search/search-input/search-input.component.scss [hunk 33]
```diff
@@ -1,13 +1,33 @@
...
 $search-border-radius: 4px;
 $top-margin: 12px;
 
 .app-search-container {
   color: var(--theme-foreground-text-color);
+  max-width: 100%;
+  margin: 0 !important;
+
+  .app-search-button {
+    width: 32px !important;
```
Comment (MichalKinas) [inline projects/aca-content/src/lib/components/search/search-input/search-input.component.scss:13]: Do we need all these `important`? You don't override anything from ADF so I think it can be achieved with them

File: projects/aca-content/src/lib/components/search/search-input/search-input.component.ts [hunk 34]
```diff
@@ -140,7 +152,6 @@ export class SearchInputComponent implements OnInit, OnDestroy {
     } else {
       this.store.dispatch(new SnackbarErrorAction('APP.BROWSE.SEARCH.EMPTY_SEARCH'));
     }
-    this.trigger.closeMenu();
```
Comment (MichalKinas) [inline projects/aca-content/src/lib/components/search/search-input/search-input.component.ts:143]: Why did you remove this?

File: projects/aca-content/src/lib/components/sidenav/components/expand-menu.component.html [hunk 35]
```diff
@@ -10,36 +10,30 @@
       mat-button
       class="action-button full-width"
     >
-      <adf-icon *ngIf="item.icon" [value]="item.icon"></adf-icon>
```
Comment (MichalKinas) [inline projects/aca-content/src/lib/components/sidenav/components/expand-menu.component.html:13]: Why we are removing icons for menu items? Is it what new design should do?

File: projects/aca-content/src/lib/components/sidenav/directives/expansion-panel.directive.ts [hunk 36]
```diff
@@ -64,6 +64,26 @@ export class ExpansionPanelDirective implements OnInit, OnDestroy {
...
+
+  isRouteProfile(): boolean {
+    return this.router.url.includes('/profile');
+  }
+
+  isRouteSearch(): boolean {
+    return this.router.url.includes('/search');
+  }
+
+  isItemExpanded() {
```
Comment (MichalKinas) [inline projects/aca-content/src/lib/components/sidenav/directives/expansion-panel.directive.ts:79]: Return type is missing

File: projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.ts [hunk 37]
```diff
@@ -0,0 +1,33 @@
...
+import { Component, OnInit, ViewEncapsulation } from '@angular/core';
+import { ContentApiService } from '@alfresco/aca-shared';
+
+@Component({
+  selector: 'app-navigation-menu',
+  templateUrl: './navigation-menu.component.html',
+  encapsulation: ViewEncapsulation.None
+})
+export class NavigationMenuComponent implements OnInit {
+  displayName = 'Unknown user';
```
Comment (MichalKinas) [inline projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.ts:18]: I think it should be provided in translation files

**Review by MichalKinas — CHANGES_REQUESTED** [review]

**Comment by MichalKinas** [comment]
image
After clicking on search sidenav looks broken

**Comment by MichalKinas** [comment]
Zrzut ekranu 2023-02-13 o 12 21 41
Also there are some errors in the console which are not there on develop

File: projects/aca-content/src/lib/components/header-actions/header-actions.component.ts [hunk 38]
```diff
@@ -0,0 +1,88 @@
...
+  }
+
+  canShowCreateButton(): boolean {
+    if (this.isPersonalFilesRoute() || this.isFavoriteLibrariesRoute() || this.isLibrariesRoute()) {
+      return true;
```
[the remark this answers is not in the document]
Comment (DenysVuika) [reply projects/aca-content/src/lib/components/header-actions/header-actions.component.ts]: updated, also made those "is-" methods private until we can provide better solution

File: projects/aca-content/src/lib/components/search/search-input/search-input.component.ts (hunk 34 above)
Comment (DenysVuika) [reply projects/aca-content/src/lib/components/search/search-input/search-input.component.ts:143]: this is strange, was thinking about the same, note that I'm just cleaning the garbage here :) will restore

File: projects/aca-content/src/lib/components/search/search-results/search-results.component.scss [hunk 39]
```diff
@@ -3,6 +3,13 @@
 $adf-chip-background: #efefef;
 $contrast-gray: #646569;
 
+.adf-toolbar-search-results {
+  position: fixed;
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/search/search-results/search-results.component.scss:7]: this is probably the most lame way of positioning... putting flex everywhere, and then making fixed position like it is 1995 now

File: projects/aca-shared/src/lib/components/page-layout/page-layout.component.scss [hunk 40]
```diff
@@ -3,14 +3,43 @@
...
+    border-right: none;
+    background: var(--theme-page-layout-header-background-color);
+    height: 96px;
+    padding: 0 24px;
+    display: flex;
+    align-items: center;
+  }
+
+  .aca-menu-icon {
+    cursor: pointer;
```
Comment (DenysVuika) [inline projects/aca-shared/src/lib/components/page-layout/page-layout.component.scss]: this is stupid, you can use mat-button-icon like everywhere else is used

File: projects/aca-content/src/lib/components/header-actions/header-actions.component.scss [hunk 41]
```diff
@@ -0,0 +1,47 @@
...
+    border: none;
+    height: 32px;
+  }
+
+  .aca-create-button {
+    background-color: var(--theme-create-button-background-color);
+    color: var(--theme-create-button-text-color);
+  }
+
+  .aca-upload-button {
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/header-actions/header-actions.component.scss:43]: this needs to be just "primary" button

File: projects/aca-content/src/lib/components/header-actions/header-actions.component.scss [hunk 42]
```diff
@@ -0,0 +1,47 @@
...
+      height: 32px;
+      margin: 4px 0 0 12px;
+    }
+
+    & > button {
+      margin-left: 12px;
+    }
+  }
+
+  .aca-mat-button {
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/header-actions/header-actions.component.scss:24]: this needs to go, something in the parent elements messes up the button layout; we should be using just material button here without custom styles

File: projects/aca-content/src/lib/components/header-actions/header-actions.component.scss [hunk 43]
```diff
@@ -0,0 +1,47 @@
...
+    font-weight: 500;
+    font-size: 14px;
+    font-style: normal;
+    border-radius: 6px;
+    border: none;
+    height: 32px;
+  }
+
+  .aca-create-button {
+    background-color: var(--theme-create-button-background-color);
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/header-actions/header-actions.component.scss:39]: should be just default button style, without this customisation

File: projects/aca-content/src/lib/components/sidenav/sidenav.component.scss [hunk 44]
```diff
@@ -1,150 +1,129 @@
...
+      display: flex;
+      width: 100%;
+    }
+
+    .action-button--active {
+      color: var(--theme-selected-text-color) !important;
+      background: var(--theme-selected-background-color);
+    }
+
+    .item:hover .action-button__label {
```
Comment (DenysVuika) [inline projects/aca-content/src/lib/components/sidenav/sidenav.component.scss:116]: should be nested with item (the author probably has no idea how scss works)

File: projects/aca-content/src/lib/components/header/header.component.spec.ts (hunk 29 above)
Comment (DenysVuika) [reply projects/aca-content/src/lib/components/header/header.component.spec.ts:107]: the whole component is removed as no longer used

File: projects/aca-content/src/lib/components/search/search-input-control/search-input-control.component.scss (hunk 30 above)
Comment (DenysVuika) [reply projects/aca-content/src/lib/components/search/search-input-control/search-input-control.component.scss:9]: seems to be needed, as some other styles messing this button
image

File: projects/aca-content/src/lib/components/search/search-input.service.ts (hunk 32 above)
Comment (DenysVuika) [reply projects/aca-content/src/lib/components/search/search-input.service.ts:32]: yeah, I will rename it to `SearchNavigationService` and maybe in the future should consider getting rid of it

File: projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.ts (hunk 37 above)
Comment (DenysVuika) [reply projects/aca-content/src/lib/components/sidenav/navigation-menu/navigation-menu.component.ts:18]: will make empty and add "ngIf" to the html template

**Review by DenysVuika — APPROVED** [review]

**Review by MohinishSah — APPROVED** [review]
LGTM

File: app/src/app.config.json.tpl [hunk 45]
```diff
@@ -32,16 +32,13 @@
   },
   "locale": "en",
   "application": {
-    "name": "Alfresco Content Application",
-    "version": "4.0.0-A.3",
-    "logo": "assets/images/alfresco-logo-flower.svg",
-    "headerImagePath": "assets/images/mastHead-bg-shapesPattern.svg",
+    "name": "Workspace",
+    "version": "4.0.0-A.2",
```
Comment (DenysVuika) [inline app/src/app.config.json.tpl:36]: should be A.3

Comment (SheenaMalhotra182) [reply app/src/app.config.json.tpl]: changed to A.3
